### PR TITLE
oil 0.7.0 (new formula)

### PR DESCRIPTION
**Pull request by waldyrious**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

**Comment by waldyrious** [comment]
@andychu, for the test block I used the "smoke test" section of https://www.oilshell.org/release/latest/doc/INSTALL.html for inspiration, but at some point during local testing, the produced AST for the demo script I was using changed as follows:

```diff
-       spids: [5 18]
+       spids: [4 17]
```

Any insights as to what could explain this change?

**Comment by andychu** [comment]
`-n` is more of an informal demo for users -- I wouldn't  use that as the test.  That format is subject to change at any time (and maybe I should make that more clear somehow)

Also despite the versioning, the later releases are almost certainly strictly better than earlier ones, so

https://www.oilshell.org/release/0.8.pre2/

is worth packaging.

Thanks for packaging Oil!  Please update the wiki when it's merged and let me know if you have more questions.

https://github.com/oilshell/oil/wiki/Oil-Deployments

**Comment by andychu** [comment]
Oh also  eventually Oil will be distributed with tests you can run.  That's this issue:

https://github.com/oilshell/oil/issues/167

If you want to help let me know :)  There's been some big progress recently at http://travis-ci.oilshell.org/jobs/ (details on Zulip)

**Comment by chenrui333** [comment]
@andychu just curious why this oilshell depends on Python 2, not Python 3. Also, depends on python 2.7.13 (not 2.7.16)

**Review by chenrui333 — APPROVED** [review]
Thanks for bring this shell into my attention, good to know there is always another one exist. :)

**Comment by andychu** [comment]
The goal is to remove the CPython dependency altogether.  Some details:

http://www.oilshell.org/blog/2020/01/parser-benchmarks.html

http://www.oilshell.org/blog/2019/06/17.html#why-is-it-written-in-python

**Comment by SMillerDev** [comment]
If it depends on Python it should define `uses_from_macos "python@2"`

**Comment by andychu** [comment]
It doesn't require Python to be installed:  From the packager's perspective it looks like a C program.

**Comment by SMillerDev** [comment]
Does it require python to run?

**Comment by waldyrious** [comment]
> -n is more of an informal demo for users -- I wouldn't use that as the test. 

What would you suggest instead, then? Perhaps one of the spec tests?

> despite the versioning, the later releases are almost certainly strictly better than earlier ones, so oilshell.org/release/0.8.pre2 is worth packaging.

Sure, I can do that. @SMillerDev, @chenrui333, is that OK from your side as well?

**Comment by andychu** [comment]
I would just remove it --  `-c "echo hi"` is  a good sanity check already.  And as mentioned Oil issue 167 is for people who want more than that.

@SMillerDev It doesn't require Python to run

**Comment by waldyrious** [comment]
> Thanks for packaging Oil! Please update the wiki when it's merged

No problem, and will do 👍

**Comment by SMillerDev** [comment]
> is that OK from your side as well?

Nope, only stable releases are allowed.

**Comment by waldyrious** [comment]
Another question: is there any way to link to the source code repository of the formula when it has a dedicated website? I know I could use `head`, but IIUC that implies that the formula should support building from the master branch, which AFAIK is not the case — according to the [README](https://github.com/oilshell/oil#quick-start-on-linux):

> Developer builds don't work on OS X, so use the release tarballs on OS X.

**Comment by SMillerDev** [comment]
You can use head for that. While it allows people to try and build the software, that configuration means that you're on your own if it breaks.

**Comment by waldyrious** [comment]
I've added the `head` block and simplified the test. I also made a minor adjustment to the head block of another formula, but I can submit that as a separate PR if it's preferable.

File: Formula/gobby.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class Gobby < Formula
   url "http://releases.0x539.de/gobby/gobby-0.5.0.tar.gz"
   sha256 "8ceb3598d27cfccdf9c9889b781c4c5c8e1731ca6beb183f5d4555644c06bd98"
   revision 7
-  head "https://github.com/gobby/gobby"
+  head "https://github.com/gobby/gobby.git"
```
Comment (SMillerDev) [inline Formula/gobby.rb]: Please make this a separate PR

Comment (waldyrious) [reply Formula/gobby.rb]: Will do 👍

**Comment by waldyrious** [comment]
Fixed the audit error and removed the change to an unrelated formula.

**Comment by SMillerDev** [comment]
If you can just squash the commits into one `<name> <version> (new formula)` now?

**Comment by waldyrious** [comment]
Sure, I wasn't sure if force-pushing was OK. Will squash now.

File: Formula/gobby.rb (hunk 1 above)
Comment (waldyrious) [reply Formula/gobby.rb]: @SMillerDev I suspect I'll need to also fix [this](https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/59678/version=catalina/testReport/junit/brew-test-bot/catalina/audit_gobby___online/) if I am to submit that change, but I'm not sure exactly how to do that.

Can you clarify what adjustment I would need to do to [gobby/0.5.0.patch](https://github.com/Homebrew/formula-patches/blob/master/gobby/0.5.0.patch), and whether that would need to be done before or after the PR in this repo?

**Comment by chenrui333** [comment]
@waldyrious Thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!
